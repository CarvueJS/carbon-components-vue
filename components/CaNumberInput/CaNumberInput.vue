<template>
  <div data-numberinput class="bx--number">
    <input id="number-input" type="number" :min="min" :max="max" :value="value">
    <div class="bx--number__controls">
      <button class="bx--number__control-btn up-icon">
        <svg viewBox="0 2 10 5" width="10" height="5" fill-rule="evenodd">
          <path d="M10 5L5 0 0 5z"></path>
        </svg>
      </button>
      <button class="bx--number__control-btn down-icon">
        <svg viewBox="0 2 10 5" width="10" height="5" fill-rule="evenodd">
          <path d="M10 0L5 5 0 0z"></path>
        </svg>
      </button>
    </div>
  </div>
</template>

<script>
  import { NumberInput } from 'carbon-components'

  export default {
    name: 'ca-number-input',
    data: () => ({
      numberInput: null,
    }),
    props: {
      value: null,
      max: {
        type: Number,
        default: 100,
      },
      min: {
        type: Number,
        default: 0,
      },
    },
    mounted () {
      this.numberInput = NumberInput.create(document.querySelector('[data-numberinput]'))
      this.numberInput.element.addEventListener('change', (e)=>{
        this.$emit('input', e.target.value)
      })
    }
  };
</script>
