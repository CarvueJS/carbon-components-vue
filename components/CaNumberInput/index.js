import CaNumberInput from './CaNumberInput.vue';

CaNumberInput.install = (Vue) => {
  Vue.component(CaNumberInput.name, CaNumberInput);
};

export default CaNumberInput;
