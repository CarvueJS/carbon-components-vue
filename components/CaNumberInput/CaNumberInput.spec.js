import { shallow } from 'vue-test-utils';
import CaNumberInput from './CaNumberInput.vue';

describe('CaNumberInput', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaNumberInput);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('Mathes snapshot', () => {
    const wrapper = shallow(CaNumberInput);
    expect(wrapper.html()).toMatchSnapshot()
  });
})
