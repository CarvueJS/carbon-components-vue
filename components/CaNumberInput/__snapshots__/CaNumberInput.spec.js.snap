// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaNumberInput Mathes snapshot 1`] = `
<div data-numberinput="" class="bx--number">
  <input id="number-input" type="number" min="0" max="100" step="10">
  <div class="bx--number__controls">
    <button class="bx--number__control-btn up-icon">
      <svg viewBox="0 2 10 5" width="10" height="5" fill-rule="evenodd">
        <path d="M10 5L5 0 0 5z"></path>
      </svg>
    </button>
    <button class="bx--number__control-btn down-icon">
      <svg viewBox="0 2 10 5" width="10" height="5" fill-rule="evenodd">
        <path d="M10 0L5 5 0 0z"></path>
      </svg>
    </button>
  </div>
</div>
`;
